Cabal/Distribution/Version.hs

```diff
@@ -516,12 +564,11 @@ simplifyVersionRange vr
 --
 
 wildcardUpperBound :: Version -> Version
-wildcardUpperBound (Version lowerBound ts) = Version upperBound ts
-  where
-    upperBound = init lowerBound ++ [last lowerBound + 1]
+wildcardUpperBound = withVersion $
+    \lowerBound -> init lowerBound ++ [last lowerBound + 1]
 
 isWildcardRange :: Version -> Version -> Bool
-isWildcardRange (Version branch1 _) (Version branch2 _) = check branch1 branch2
+isWildcardRange ver1 ver2 = check (unVersion ver1) (unVersion ver2)
```

Comment:
I think this is an example case where arguably it'd read nicer / more clearly as

    isWildcardRange ver1 ver2 = check (versionDigits ver1) (versionDigits ver2)